diff --git a/features/bootstrap/FeatureContext.php b/features/bootstrap/FeatureContext.php
index b3299b4..1662a52 100644
--- a/features/bootstrap/FeatureContext.php
+++ b/features/bootstrap/FeatureContext.php
@@ -21,6 +21,20 @@ abstract class HackyDataRegistry {
   }
 }
 
+class LocalDataRegistry {
+  public $data = array();
+  public function set($name, $value) {
+    $this->data[$name] = $value;
+  }
+  public function get($name) {
+    $value = "";
+    if (isset($this->data[$name])) {
+      $value = $this->data[$name];
+    }
+    return $value;
+  }
+}
+
 use Behat\Behat\Exception\PendingException,
     Behat\Gherkin\Node\TableNode;
 use Drupal\DrupalExtension\Context\DrupalContext;
@@ -65,6 +79,13 @@ class FeatureContext extends DrupalContext {
    * Store the file name of a downloaded file
    */
   private $downloadedFileName = '';
+
+  /**
+   * Create a context specific data storage container.
+   */
+
+  private $dataRegistry = '';
+
   /**
    * Initializes context.
    *
@@ -74,6 +95,7 @@ class FeatureContext extends DrupalContext {
    *   Context parameters (set them up through behat.yml or behat.local.yml).
    */
   public function __construct(array $parameters) {
+    $this->dataRegistry = new LocalDataRegistry();
     $this->default_browser = $parameters['default_browser'];
     if (isset($parameters['drupal_users'])) {
       $this->drupal_users = $parameters['drupal_users'];
@ -4159,10 +4209,10 @@ class FeatureContext extends DrupalContext {
       $title = $this->issueTitle;
       $type = 'Issue';
     }
-    elseif ($title = HackyDataRegistry::get('book page title')) {
+    elseif ($title = $this->dataRegistry->get('book page title')) {
       $type = 'Document';
     }
-    elseif ($title = HackyDataRegistry::get('random:Forum subject')) {
+    elseif ($title = $this->dataRegistry->get('random:Forum subject')) {
       $type = 'Forum';
     }
     if (empty($title) || empty($element) || strpos($element->getText(), $title) === FALSE) {
@@ -6093,11 +6143,11 @@ class FeatureContext extends DrupalContext {
    * @When /^I create(?:d|) a forum(?:| topic)$/
    */
   public function iCreateAForum() {
-    sleep(3);
+    // sleep(3);
     $page = $this->getSession()->getPage();
     $subject = $this->randomString(8);
     $page->fillField("Subject", $subject);
-    HackyDataRegistry::set('random:Forum subject', $subject);
+    $this->dataRegistry->set('random:Forum subject', $subject);
     $summary = str_repeat($this->randomString(10) . " ", 10);
     // Fill summary
      // If javascript is used, then click Edit summary link and then fill field
@@ -6112,15 +6162,26 @@ class FeatureContext extends DrupalContext {
         $summary_ele->setValue($summary);
       }
     }
-    HackyDataRegistry::set('random:Forum summary', $summary);
+    $this->dataRegistry->set('random:Forum summary', $summary);
     $body = str_repeat($this->randomString(30) . " ", 10);
     $page->fillField("Body", $body);
-    HackyDataRegistry::set('random:Forum body', $body);
+    $this->dataRegistry->set('random:Forum body', $body);
     $page->pressButton('Save');
-    // Let the page load
-    sleep(3);
     // Store node url
-    HackyDataRegistry::set('forum url', $this->getSession()->getCurrentUrl());
+      $this->dataRegistry->set('forum url', $this->getSession()->getCurrentUrl());
+  }
+
+  /**
+   * @Given /^there is a new "([^"]*)" forum topic$/
+   */
+  public function thereIsANewForumTopic($forum) {
+    return array (
+      new Given("I am logged in as \"site user\""),
+      new Given("I am at \"/node/add/forum/0\""),
+      new Given("I select \"-$forum\" from \"edit-taxonomy-forums-und\""),
+      new Given("I create a forum topic"),
+      // User should be logged out after data set up
+    );
   }
 
   /**
@@ -6142,11 +6203,11 @@ class FeatureContext extends DrupalContext {
    * @Then /^I should see the (?:community spotlight|forum topic) link$/
    */
   public function iShouldSeeTheForumLink() {
-    if (!($subject = HackyDataRegistry::get('random:Forum subject'))) {
+    if (!($subject = $this->dataRegistry->get('random:Forum subject'))) {
       throw new Exception('Forum subject is empty');
     }
     // Let the page load
-    sleep(3);
+    //  sleep(3);
     return new Then('I should see the link "' . $subject . '"');
   }
 
@@ -6177,7 +6238,7 @@ class FeatureContext extends DrupalContext {
     // Remove read more from the intro
     $intro = trim(str_replace("Read more", "", $intro));
     // Get the summary from post and check if the intro is part of it or not
-    if (strpos(HackyDataRegistry::get('random:Forum summary'), $intro) === FALSE) {
+    if (strpos($this->dataRegistry->get('random:Forum summary'), $intro) === FALSE) {
       throw new Exception('The news section did not contain summary text');
     }
   }
@@ -6510,7 +6571,7 @@ class FeatureContext extends DrupalContext {
    */
   public function iShouldSeeLatestForumTopicInTheRightsideBlock() {
     sleep(6);
-    $forumTitle = HackyDataRegistry::get('random:Forum subject');
+    $forumTitle = $this->dataRegistry->get('random:Forum subject');
     if(empty($forumTitle)) {
       throw new Exception('No Forum title exists in this page');
     }
diff --git a/features/drupalorg/frontpage_forum_posts.feature b/features/drupalorg/frontpage_forum_posts.feature
index 870a2d0..585ffb6 100644
--- a/features/drupalorg/frontpage_forum_posts.feature
+++ b/features/drupalorg/frontpage_forum_posts.feature
@@ -4,7 +4,6 @@ Feature: Forum posts on front page
   As any user
   I should be able to view the Forum Posts tab on the Drupal front page
 
-  @api
   Scenario: Create a forum post
     Given I am logged in as "site user"
     And I visit "/forum"
@@ -12,28 +11,25 @@ Feature: Forum posts on front page
     And I follow "Add new Forum topic"
     When I create a forum topic
     And I see "has been created"
-    #And I visit "/cron.php?cron_key=drupal"
-    And the cache has been cleared
     And I visit "/forum"
     And I follow "News and announcements"
     Then I should see the forum topic link
-
-  @dependent @anon @flaky @javascript
+    
+  @javascript
   Scenario: Forum Posts tab on front page
-    Given I am on the homepage
+    Given there is a new "Paid Drupal services" forum topic
+    And I am on the homepage
     When I follow "Forum Posts"
     Then I should see the forum topic link
     And I should see "Posted by"
     And I should see at least "5" links under the "Forum Posts" tab
     And I should see the link "More forums"
 
-  @anon @flaky @clean_data
   Scenario: Forum Posts tab on front page: More
-    Given I am on the homepage
+    Given there is a new "Post installation" forum topic
+    And I am on the homepage
     When I follow "Forum Posts"
     And I follow "More forums"
-    And I follow "News and announcements"
-    Then I should see "Community"
-    And I should see the heading "News and announcements"
+    And I follow "Post installation"
+    Then I should see the heading "Post installation"
     And I should see the forum topic link
-    And I should see "Log in to post new content in the forum"
